data-prepper-plugins/otel-trace-source/src/main/java/com/amazon/dataprepper/plugins/certificate/acm/ACMCertificateProvider.java

```diff
@@ -0,0 +1,134 @@
+    private static final long SLEEP_INTERVAL = 10000L;
+    private static final String BOUNCY_CASTLE_PROVIDER = "BC";
+    private final AWSCertificateManager awsCertificateManager;
+    private final long totalTimeout;
+
+    public ACMCertificateProvider(final String awsRegion, final long totalTimeout) {
+        final AWSCredentialsProvider credentialsProvider = new DefaultAWSCredentialsProviderChain();
+        final ClientConfiguration clientConfig = new ClientConfiguration()
+                .withThrottledRetries(true);
+        this.awsCertificateManager = AWSCertificateManagerClientBuilder.standard()
+            .withRegion(awsRegion)
+            .withCredentials(credentialsProvider)
+            .withClientConfiguration(clientConfig)
+            .build();
+        this.totalTimeout = totalTimeout;
+    }
+
+    // accessible only in the same package for unit test
+    ACMCertificateProvider(final AWSCertificateManager awsCertificateManager, final long totalTimeout) {
+        this.awsCertificateManager = awsCertificateManager;
+        this.totalTimeout = totalTimeout;
+    }
+
+    public Certificate getACMCertificate(final String acmArn, final String passphrase) {
+        ExportCertificateResult exportCertificateResult = null;
+        long timeSlept = 0L;
+        while (exportCertificateResult == null && timeSlept < totalTimeout) {
+            try {
+                final ExportCertificateRequest exportCertificateRequest = new ExportCertificateRequest()
+                        .withCertificateArn(acmArn)
+                        .withPassphrase(ByteBuffer.wrap(passphrase.getBytes()));
+                exportCertificateResult = awsCertificateManager.exportCertificate(exportCertificateRequest);
+
+            } catch (final RequestInProgressException ex) {
+                try {
+                    Thread.sleep(SLEEP_INTERVAL);
+                } catch (InterruptedException iex) {
+                    throw new RuntimeException(iex);
+                }
+            } catch (final ResourceNotFoundException | InvalidArnException ex) {
+                LOG.error("Exception retrieving the certificate with arn: {}", acmArn, ex);
+                throw ex;
+            }
+            timeSlept += SLEEP_INTERVAL;
+        }
+        if(exportCertificateResult != null) {
+            final String decryptedPrivateKey = getDecryptedPrivateKey(exportCertificateResult.getPrivateKey(), passphrase);
+            return new Certificate(exportCertificateResult.getCertificate(), decryptedPrivateKey);
+        } else {
+            throw new IllegalStateException(String.format("Exception retrieving certificate results. Time spent retrieving certificate is %d ms and total time out set is %d ms.", timeSlept, totalTimeout));
+        }
+    }
+
+    private String getDecryptedPrivateKey(final String encryptedPrivateKey, final String keyPassword) {
+        Security.addProvider(new BouncyCastleProvider());
```

Comment:
Does it matter that `addProvider` could be called multiple times (if this method is called multiple times?)

Reply:
Will move this to constructor.